internal/bminventory/inventory.go

```diff
@@ -1990,24 +1994,32 @@ func (b *bareMetalInventory) updateNetworks(db *gorm.DB, params installer.V2Upda
 			}
 		}
 		cluster.ClusterNetworks = params.ClusterUpdateParams.ClusterNetworks
+		updated = true
 	}
 
-	if params.ClusterUpdateParams.ServiceNetworks != nil {
+	if params.ClusterUpdateParams.ServiceNetworks != nil && !network.AreServiceNetworksIdentical(params.ClusterUpdateParams.ServiceNetworks, cluster.ServiceNetworks) {
 		for _, serviceNetwork := range params.ClusterUpdateParams.ServiceNetworks {
 			if err = network.VerifyClusterOrServiceCIDR(string(serviceNetwork.Cidr)); err != nil {
 				return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, "Service network CIDR %s", string(serviceNetwork.Cidr)))
 			}
 		}
 		cluster.ServiceNetworks = params.ClusterUpdateParams.ServiceNetworks
+		updated = true
 	}
 
-	if params.ClusterUpdateParams.MachineNetworks != nil {
+	if params.ClusterUpdateParams.MachineNetworks != nil && !network.AreMachineNetworksIdentical(params.ClusterUpdateParams.MachineNetworks, cluster.MachineNetworks) {
 		for _, machineNetwork := range params.ClusterUpdateParams.MachineNetworks {
 			if err = network.VerifyMachineCIDR(string(machineNetwork.Cidr), common.IsSingleNodeCluster(cluster)); err != nil {
 				return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, "Machine network CIDR '%s'", string(machineNetwork.Cidr)))
 			}
 		}
 		cluster.MachineNetworks = params.ClusterUpdateParams.MachineNetworks
+		updated = true
+	}
+
+	if swag.BoolValue(params.ClusterUpdateParams.VipDhcpAllocation) != swag.BoolValue(cluster.VipDhcpAllocation) ||
+		swag.BoolValue(params.ClusterUpdateParams.UserManagedNetworking) != swag.BoolValue(cluster.UserManagedNetworking) {
```

Comment:
Where are these updated?
Did we just need to add this because the `updated` variable was also added?

Reply:
Because of this https://github.com/openshift/assisted-service/blob/ff812ad654f743c9c9a044c8b1e8cb6aca042d44/internal/bminventory/inventory.go#L2161

We need to verify that the machine networks are updated in the database